src/main/java/com/minecolonies/coremod/colony/buildings/BuildingWonder.java

```diff
@@ -0,0 +1,114 @@
+package com.minecolonies.coremod.colony.buildings;
+
+import com.google.common.collect.ImmutableList;
+import com.ldtteam.blockout.views.Window;
+import com.minecolonies.api.blocks.ModBlocks;
+import com.minecolonies.api.colony.ICitizenData;
+import com.minecolonies.api.colony.IColony;
+import com.minecolonies.api.colony.IColonyView;
+import com.minecolonies.api.colony.buildings.IWonder;
+import com.minecolonies.api.colony.buildings.ModBuildings;
+import com.minecolonies.api.colony.buildings.registry.BuildingEntry;
+import com.minecolonies.api.colony.buildings.workerbuildings.IWareHouse;
+import com.minecolonies.api.colony.jobs.IJob;
+import com.minecolonies.api.crafting.ItemStorage;
+import com.minecolonies.api.entity.citizen.Skill;
+import com.minecolonies.coremod.client.gui.WindowHutComposter;
+import com.minecolonies.coremod.client.gui.WindowHutWonder;
+import com.minecolonies.coremod.colony.buildings.AbstractBuilding;
+import com.minecolonies.coremod.colony.buildings.AbstractFilterableListBuilding;
+import com.minecolonies.coremod.colony.buildings.views.AbstractFilterableListsView;
+import com.minecolonies.coremod.colony.jobs.JobComposter;
+import com.minecolonies.coremod.research.ResearchInitializer;
+import com.minecolonies.coremod.research.UnlockBuildingResearchEffect;
+import net.minecraft.block.Block;
+import net.minecraft.entity.player.PlayerEntity;
+import net.minecraft.nbt.CompoundNBT;
+import net.minecraft.nbt.ListNBT;
+import net.minecraft.nbt.NBTUtil;
+import net.minecraft.network.PacketBuffer;
+import net.minecraft.util.Tuple;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.text.TranslationTextComponent;
+import net.minecraft.world.World;
+import net.minecraftforge.common.util.Constants;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class BuildingWonder extends AbstractBuilding implements IWonder
+{
+    private static final String WONDER = "wonder";
+
+    /**
+     * Maximum building level
+     */
+    private static final int MAX_BUILDING_LEVEL = 5;
+
+    /**
+     * The constructor of the building.
+     *
+     * @param c the colony
+     * @param l the position
+     */
+    public BuildingWonder(@NotNull final IColony c, final BlockPos l)
+    {
+        super(c, l);
+    }
+
+    @NotNull
+    @Override
+    public String getSchematicName()
+    {
+        return WONDER;
+    }
+
+    @Override
+    public int getMaxBuildingLevel()
+    {
+        return MAX_BUILDING_LEVEL;
+    }
+
+    @Override
+    public BuildingEntry getBuildingRegistryEntry()
+    {
+        return ModBuildings.wonder;
+    }
+
+    /**
+     * The client side representation of the building.
+     */
+    public static class View extends AbstractFilterableListsView
```

Comment:
This doesn't have to be FitlerableListsView. Since it doesn't have a special UI.

Reply:
Updated to extends AbstractBuildingView instead